Dockerfile

```
@@ -21,45 +21,48 @@ RUN ./install-base-packages.sh && rm ./install-base-packages.sh
 FROM base-image AS install-image
 
 # Install uv.
-COPY --from=ghcr.io/astral-sh/uv:0.6.5 /uv /bin/uv
+COPY --from=ghcr.io/astral-sh/uv:0.6.17 /uv /bin/uv
 
 # Install system packages only needed for building dependencies.
 COPY scripts/install-dependency-packages.sh .
-RUN ./install-dependency-packages.sh
-
-# Create a Python virtual environment
-ENV VIRTUAL_ENV=/opt/venv
-RUN uv venv $VIRTUAL_ENV
-
-# Make sure we use the virtualenv
-ENV PATH="$VIRTUAL_ENV/bin:$PATH"
-
-# Install the app's Python runtime dependencies
-COPY requirements/main.txt ./requirements.txt
-RUN uv pip install --compile-bytecode --verify-hashes --no-cache \
-    -r requirements.txt
-
-# Install the application.
-COPY . /workdir
-WORKDIR /workdir
-RUN uv pip install --compile-bytecode --no-cache .
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    ./install-dependency-packages.sh
+
+# Disable hard links during uv package installation since we're using a
+# cache on a separate file system.
+ENV UV_LINK_MODE=copy
+
+# Install the dependencies.
+WORKDIR /app
+RUN --mount=type=cache,target=/root/.cache/uv \
+    --mount=type=bind,source=uv.lock,target=uv.lock \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    uv sync --frozen --no-default-groups --compile-bytecode --no-install-project
+
+# Install the Noteburst Python application.
+ADD . /app
+RUN --mount=type=cache,target=/.root/.cache/uv \
```

```suggestion
RUN --mount=type=cache,target=/root/.cache/uv \
```

Double check this; I think `/root` was right when we installed dependencies above.

---

Oof, great catch.

---

Oof, great catch.

---

Oof, great catch.